backend/src/main/java/com/htecgroup/skynest/service/impl/PermissionServiceImpl.java

```diff
@@ -237,6 +237,21 @@ public void revokePermission(UUID bucketId, UUID userId) {
     permissionRepository.delete(permission);
   }
 
+  @Override
+  public void revokeFilePermission(UUID fileId, UUID userId) throws FileNotFoundException {
+    FileMetadataEntity fileMetadataEntity =
+        fileRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+    checkIfBucketIsDeleted(fileMetadataEntity.getBucket().getId());
+    currentUserHasPermissionForFile(fileMetadataEntity, AccessType.OWNER);
+    currentUserHasPermissionForFolder(fileMetadataEntity.getParentFolder(), AccessType.OWNER);
+    currentUserHasPermissionForBucket(fileMetadataEntity.getBucket().getId(), AccessType.OWNER);
```

Comment:
I don't think these last two checks are necessary
`currentUserHasPermissionForFile()` should be enough